src/java.base/share/classes/java/nio/file/Files.java

```diff
@@ -1569,7 +1570,8 @@ public static boolean isSameFile(Path path, Path path2) throws IOException {
      * {@code mismatch(f,f)} returns {@code -1L}). If the file system and files
      * remain static, then this method is <i>symmetric</i> (for two {@code Paths f}
      * and {@code g}, {@code mismatch(f,g)} will return the same value as
-     * {@code mismatch(g,f)}).
+     * {@code mismatch(g,f)}). The only case where there is no file system
+     * access is when the paths are equal.
```

Comment:
What would you think of putting this in a new paragraph with "If both Path objects are equal then this method returns true without checking if the file exists"? This would be consistent with isSameFlie?

The updated description for the security exceptions looks okay.

Reply:
Sounds reasonable.

Reply:
So changed in 46384485ca6ccc0d2aab7d3fc1a20b5976d88425.